velox/type/TimestampConversion.cpp

```diff
@@ -144,6 +145,53 @@ bool isValidWeekDate(int32_t weekYear, int32_t weekOfYear, int32_t dayOfWeek) {
   return true;
 }
 
+bool isValidWeekOfMonthDate(
+    int32_t year,
+    int32_t month,
+    int32_t weekOfMonth,
+    int32_t dayOfWeek) {
+  if (year < 1 || year > kMaxYear) {
+    return false;
+  }
+  if (month < 1 || month > 12) {
+    return false;
+  }
+
+  int64_t daysSinceEpochOfFirstDayOfMonth;
+  const Status status =
+      daysSinceEpochFromDate(year, month, 1, daysSinceEpochOfFirstDayOfMonth);
+  if (!status.ok()) {
+    return false;
+  }
+
+  // Calculates the actual number of week of month and validates if it is in the
+  // valid range.
+  const int32_t firstDayOfWeek =
+      extractISODayOfTheWeek(daysSinceEpochOfFirstDayOfMonth);
+  const int32_t firstWeekLength = 7 - firstDayOfWeek + 1;
+  const int32_t monthLength =
+      isLeapYear(year) ? kLeapDays[month] : kNormalDays[month];
+  const int32_t actualWeeks = 1 + ceil((monthLength - firstWeekLength) / 7.0);
+  if (weekOfMonth < 1 || weekOfMonth > actualWeeks) {
```

Comment:
> weekOfMonth > actualWeeks

The documentation suggested that valid range for week is fixed: [1, 6], but this code suggests that the range depends on the month: [1, <number of weeks in a month>]. CC: @rui-mo

Reply:
6 is the max value of week for all months. Maybe I should change the documentation. Thanks.